lua/starfall/libs_sh/builtins.lua

```diff
@@ -201,6 +201,44 @@ function SF.DefaultEnvironment.getLibraries ()
 	return ret
 end
 
+--- Used the same as Lua's pairs but iterates over the given library's or type's default fields
+-- @param typ The Starfall Library or Type to iterate over
+-- @usage
+-- \-- Default Environment, would also work with getfenv()
+--
+-- for k, v in iter( _G ) do
+-- \   -- CODE
+-- end
+--
+-- \-- or for example:
+--
+-- for k, v in iter( render ) do
+-- \   -- CODE
+-- end
+function SF.DefaultEnvironment.iter ( typ )
+	SF.CheckType( typ, "table" )
+
+	local function checkType ( typ )
```

Comment:
This doesn't need to be a locally defined function within the iter function. This doesn't need closure, and will hamper performance.

I propose the following local function above this function definition:
```lua
--- Local type checking function for the iter function.
-- @param typ The Starfall Library or Type to check.
-- @throws Error should the input typ not be a valid Starfall Library or Type.
local function iterCheckType ( typ )
	local meta = dgetmeta( typ )
	if not (
	meta and meta.__metatable
			and type( meta.__metatable ) == "string" and SF.GetTypeDef( meta.__metatable )
			and meta.__realType and meta.__realType == SF.GetTypeDef( meta.__metatable ).__realType
	) then
		SF.throw( "iter: Invalid Library or Type", 3 )
	end
end
```

Ensure to then change checkType references to the correct iter version.

Then this should be good to go!